app/src/main/java/zapsolutions/zap/transactionHistory/transactionDetails/OnChainTransactionDetailBSDFragment.java

```diff
@@ -0,0 +1,209 @@
+        mAmountLabel.setText(amountLabel);
+        String feeLabel = getActivity().getResources().getString(R.string.fee) + ":";
+        mFeeLabel.setText(feeLabel);
+        String dateLabel = getActivity().getResources().getString(R.string.date) + ":";
+        mDateLabel.setText(dateLabel);
+        String transactionIDLabel = getActivity().getResources().getString(R.string.transactionID) + ":";
+        mTransactionIDLabel.setText(transactionIDLabel);
+        String addressLabel = getActivity().getResources().getString(R.string.address) + ":";
+        mAddressLabel.setText(addressLabel);
+
+        DateFormat df = DateFormat.getDateInstance(DateFormat.LONG, getActivity().getResources().getConfiguration().locale);
+        String formattedDate = df.format(new Date(mTransaction.getTimeStamp() * 1000L));
+
+        DateFormat tf = DateFormat.getTimeInstance(DateFormat.MEDIUM, getActivity().getResources().getConfiguration().locale);
+        String formattedTime = tf.format(new Date(mTransaction.getTimeStamp() * 1000L));
+        mDate.setText(formattedDate + ", " + formattedTime);
+
+
+        mAddress.setText(mTransaction.getDestAddresses(0));
+        mTransactionID.setText(mTransaction.getTxHash());
+
+        mTransactionID.setOnClickListener(view -> BlockExplorer.showTransaction(mTransaction.getTxHash(), getActivity()));
+        mAddress.setOnClickListener(view -> BlockExplorer.showAddress(mTransaction.getDestAddresses(0), getActivity()));
+        mTransactionIDCopyButton.setOnClickListener(view -> ClipBoardUtil.copyToClipboard(getContext(), "TransactionID", mTransaction.getTxHash()));
+        mAddressCopyButton.setOnClickListener(view -> ClipBoardUtil.copyToClipboard(getContext(), "Address", mTransaction.getDestAddresses(0)));
+
+
+        // is internal?
+        if (Wallet.getInstance().isTransactionInternal(mTransaction)) {
+            bindInternal();
+        } else {
+            bindNormalTransaction();
+        }
+    }
+
+    @Override
+    public int getTheme() {
+        return R.style.ZapBottomSheetDialogTheme;
+    }
+
+    private void bindInternal() {
+        mTransactionDescription.setText(R.string.channel_event);
+        Long amount = mTransaction.getAmount();
+
+        mAddress.setVisibility(View.GONE);
+        mAddressLabel.setVisibility(View.GONE);
+        mAddressCopyButton.setVisibility(View.GONE);
+        mAmount.setVisibility(View.GONE);
+        mAmountLabel.setVisibility(View.GONE);
+
+        if (mTransaction.getTotalFees() > 0) {
+            mFee.setText(MonetaryUtil.getInstance().getPrimaryDisplayAmountAndUnit(mTransaction.getTotalFees()));
+        } else {
+            mFee.setVisibility(View.GONE);
+            mFeeLabel.setVisibility(View.GONE);
+        }
+
+        switch (amount.compareTo(0L)) {
+            case 0:
+                // amount = 0
+                String alias = Wallet.getInstance().getNodeAliasFromChannelTransaction(mTransaction, getActivity());
```

Comment:
I guess we can move the alias and `mChannel.setText(alias);` outside of the switch.